File: Medium/38. Count and Say.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        String base = countAndSay(n - 1);
        StringBuilder sb = new StringBuilder();
        int s = 1;
        for (int i = 1; i < base.length(); i++) {
            if (base.charAt(i) != base.charAt(i - 1)) {
                sb.append(s);
                sb.append(base.charAt(i - 1));
                s = 1;
            } else {
                s++;
            }
        }
        sb.append(s);
        sb.append(base.charAt(base.length() - 1));
        return sb.toString();
    }
}

File: Medium/38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        result = '1'
        for _ in range(n - 1):
            RLE = []
            prev = result[0]
            count = 1
            for i in range(1, len(result)):
                if result[i] == prev:
                    count += 1
                else:
                    RLE.append(str(count))
                    RLE.append(prev)
                    prev = result[i]
                    count = 1
            RLE.append(str(count))
            RLE.append(prev)
            result = ''.join(RLE)
        return result